### DISCUSSION TITLE
Is there a way to prevent delete block on backspace?

### DISCUSSION THREAD

**Discussion by CavalcanteLeo** [codex-team/editor.js#1906] [Q&A]
Hi, i'm creating a platform for radiologists that makes structures reports.
So, users **can not** remove any block. 

But when i press backspace, it deletes the whole block. Is there a way to prevent it? Also i can not allow to delete multiple blocks by selecting them

Users are still allow to delete letters/words from the block, but not the block itself

**Comment by gaona233** [comment]
How did you solve it，i have same problem

**Comment by crying2812** [comment]
Hi @CavalcanteLeo, can you show me the full of sourcecode? I'm a newbie, i cant find where i can put your code

**Comment by MakarovCode** [comment]
Any way to solve this issue?

**Comment by conanliuhuan** [comment]
Any way to solve this issue?
If i rendered a custom-web-component with shadow-root, backspace in that input will remove all the block

**Comment by gonzalezfj-t** [comment]
I have a hacky workaround that solved the problem for me. 

Context:  A custom plugin made with react, a simple gallery of pictures, created with one wrapper div,  an input to upload, and a dynamically added set of image tags to show the pictures. The problem is similar, I need to prevent someone from accidentally removing a gallery full of pictures by pressing the backspace, like to the behavior of the native "imageTool". 
Showing the code I encountered that: 
https://github.com/codex-team/editor.js/blob/ff91466b148eb58f935c4449bf700a627a785d79/src/components/modules/blockEvents.ts#L277
The backspace event only removes blocks that are empty checking the property isEmpty. But the getter "isEmpty" does not consider nested input or images:
https://github.com/codex-team/editor.js/blob/6c0555a3229a16641709cdc83ec248af87f3c77a/src/components/block/index.ts#L411

TLDR: As a workaround, I put a `<input hidden type="text">` at the top level of the plugin.  (hidden attribute no type)

**Reply by theophilhenry** [reply]
Nice one!
